src/test/chain_tests.cpp

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2011-2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <boost/test/unit_test.hpp>
+#include <chain.h>
+#include <test/util/setup_common.h>
+
+BOOST_AUTO_TEST_SUITE(chain_tests)
+
+BOOST_AUTO_TEST_CASE( get_median_time_past )
+{
+    CBlockIndex blocks[11];
+    blocks[0].nTime=0;
+    for (int i=1; i<11;i++)
```

Comment:
- prefer prefix iterator per doc/developer-notes.md 
```suggestion
    for (int i = 1; i < 11; ++i) {
```

- please run clang-format on your changes

```diff
-BOOST_AUTO_TEST_CASE( get_median_time_past )
+BOOST_AUTO_TEST_CASE(get_median_time_past)
 {
     CBlockIndex blocks[11];
-    blocks[0].nTime=0;
-    for (int i=1; i<11;i++)
-    {
+    blocks[0].nTime = 0;
+    for (int i = 1; i < 11; i++) {
         blocks[i].nTime = i;
-        blocks[i].pprev = &blocks[i-1];
+        blocks[i].pprev = &blocks[i - 1];
     }
     // block nTime already sorted in increasing order
-    BOOST_CHECK( blocks[10].GetMedianTimePast() == blocks[5].nTime);
+    BOOST_CHECK(blocks[10].GetMedianTimePast() == blocks[5].nTime);
 }
```

Reply:
I followed your advice to check the unit test code coverage for chain.h. 
chain.h::ToString() and GetBlockHeader()  are not covered by unit test.

I would like to add new test cases about ToString and GetBlockHeader() to chain_tests.cpp

I also checked existing tests about GetMedianPastTime. Some tests check that the MedianPastTime is within certain bounds; but none of the existing tests is very precise about the expected value of the median past time.